### ISSUE TITLE
Copy From with Chown on Multistage does not chown

### ISSUE THREAD

**Issue by kferrone** [GoogleContainerTools/kaniko#579]
**Actual behavior**
When I use a multi stage build which copies from the first stage to the second with a `--chown`, the directories are not actually `chowned`. 

**Expected behavior**
The chown on a `copy` should actually chown to the selected user. 

**To Reproduce**
1. Use this simple NodeJS DockerFile with any node project. 

```
FROM node:10.11.0 as builder

WORKDIR /home/node

COPY . .

RUN npm install --only=production --verbose

FROM node:10.11.0-alpine as runner

ENV NPM_CONFIG_LOGLEVEL=info \
	NODE_ENV=production

WORKDIR /home/node

COPY --from=builder --chown=node:node /home/node .

EXPOSE 3500

USER node

RUN mkdir -p logs

CMD [ "npm", "start" ]
```

2. When you build this with docker, everything works. When built with kaniko, the line with `RUN mkdir -p logs` will fail with permission denied and the whole `/home/node` dir is owned by root still. 

**Additional Information**
 
Here is what I had to do to the Dockerfile to make it work in the mean time:

```
FROM node:10.11.0 as builder

WORKDIR /home/node

COPY . .

RUN npm install --only=production --verbose

FROM node:10.11.0-alpine as runner

ENV NPM_CONFIG_LOGLEVEL=info \
	NODE_ENV=production

WORKDIR /home/node

COPY --from=builder --chown=node:node /home/node .

EXPOSE 3500

RUN chown -R node:node /home/node && \
	mkdir -p logs

USER node

CMD [ "npm", "start" ]
```

**Comment by cvgw** [comment]
related issue https://github.com/GoogleContainerTools/kaniko/issues/57

**Comment by tejal29** [comment]
@kferrone I verified your dockerfile on #962 
```
INFO[0000] Resolved base name node:10.11.0 to node:10.11.0 
INFO[0000] Resolved base name node:10.11.0-alpine to node:10.11.0-alpine 
INFO[0000] Using dockerignore file: /workspace/.dockerignore 
INFO[0000] Resolved base name node:10.11.0 to node:10.11.0 
INFO[0000] Resolved base name node:10.11.0-alpine to node:10.11.0-alpine 
INFO[0000] Retrieving image manifest node:10.11.0       
INFO[0001] Retrieving image manifest node:10.11.0       
INFO[0002] Retrieving image manifest node:10.11.0-alpine 
INFO[0002] Retrieving image manifest node:10.11.0-alpine 
INFO[0003] Built cross stage deps: map[0:[/home/node]]  
INFO[0003] Retrieving image manifest node:10.11.0       
INFO[0004] Retrieving image manifest node:10.11.0       
INFO[0004] Unpacking rootfs as cmd COPY . . requires it. 
INFO[0015] Taking snapshot of full filesystem...        
INFO[0020] WORKDIR /home/node                           
INFO[0020] cmd: workdir                                 
INFO[0020] Changed working directory to /home/node      
INFO[0021] COPY . .                                     
INFO[0021] Taking snapshot of files...                  
INFO[0032] RUN npm install --only=production --verbose  
INFO[0032] cmd: /bin/sh                                 
INFO[0032] args: [-c npm install --only=production --verbose] 
npm info it worked if it ends with ok
npm verb cli [ '/usr/local/bin/node',
npm verb cli   '/usr/local/bin/npm',
npm verb cli   'install',
npm verb cli   '--only=production',
npm verb cli   '--verbose' ]
npm info using npm@6.4.1
npm info using node@v10.11.0
npm verb npm-session b959fd2aebdfb558
npm timing stage:loadCurrentTree Completed in 8ms
npm timing stage:loadIdealTree:cloneCurrentTree Completed in 0ms
npm timing stage:loadIdealTree:loadShrinkwrap Completed in 1ms
npm timing stage:loadIdealTree:loadAllDepsIntoIdealTree Completed in 2ms
npm timing stage:loadIdealTree Completed in 4ms
npm timing stage:generateActionsToTake Completed in 2ms
npm verb correctMkdir /root/.npm/_locks correctMkdir not in flight; initializing
npm verb makeDirectory /root/.npm/_locks creation not in flight; initializing
npm timing audit compress Completed in 2ms
npm info audit Submitting payload of 115bytes
npm verb lock using /root/.npm/_locks/staging-b947fdc4509dbfe5.lock for /home/node/node_modules/.staging
npm verb unlock done using /root/.npm/_locks/staging-b947fdc4509dbfe5.lock for /home/node/node_modules/.staging
npm timing stage:executeActions Completed in 124ms
npm timing stage:rollbackFailedOptional Completed in 0ms
npm info linkStuff !invalid#1
npm info lifecycle undefined~install: undefined
npm info lifecycle undefined~postinstall: undefined
npm timing stage:runTopLevelLifecycles Completed in 152ms
npm WARN saveError ENOENT: no such file or directory, open '/home/node/package.json'
npm info lifecycle undefined~preshrinkwrap: undefined
npm info lifecycle undefined~shrinkwrap: undefined
npm notice created a lockfile as package-lock.json. You should commit this file.
npm info lifecycle undefined~postshrinkwrap: undefined
npm WARN enoent ENOENT: no such file or directory, open '/home/node/package.json'
npm verb enoent This is related to npm not being able to find a file.
npm verb enoent 
npm WARN node No description
npm WARN node No repository field.
npm WARN node No README data
npm WARN node No license field.

npm timing audit submit Completed in 256ms
npm http fetch POST 200 https://registry.npmjs.org/-/npm/v1/security/audits/quick 174ms
npm timing audit body Completed in 1ms
up to date in 0.286s
found 0 vulnerabilities

npm verb exit [ 0, true ]
npm timing npm Completed in 561ms
npm info ok 
INFO[0033] Taking snapshot of full filesystem...        
INFO[0037] Saving file /home/node for later use         
INFO[0037] Deleting filesystem...                       
INFO[0038] Retrieving image manifest node:10.11.0-alpine 
INFO[0039] Retrieving image manifest node:10.11.0-alpine 
INFO[0040] Unpacking rootfs as cmd COPY --from=builder --chown=node:node /home/node . requires it. 
INFO[0041] Taking snapshot of full filesystem...        
INFO[0042] ENV NPM_CONFIG_LOGLEVEL=info 	NODE_ENV=production 
INFO[0042] WORKDIR /home/node                           
INFO[0042] cmd: workdir                                 
INFO[0042] Changed working directory to /home/node      
INFO[0042] COPY --from=builder --chown=node:node /home/node . 
INFO[0042] Taking snapshot of files...                  
INFO[0053] EXPOSE 3500                                  
INFO[0053] cmd: EXPOSE                                  
INFO[0053] Adding exposed port: 3500/tcp                
INFO[0053] USER node                                    
INFO[0053] cmd: USER                                    
INFO[0053] RUN mkdir -p logs                            
INFO[0053] cmd: /bin/sh                                 
INFO[0053] args: [-c mkdir -p logs]                     
INFO[0053] Taking snapshot of full filesystem...        
INFO[0054] CMD [ "npm", "start" ]   
```

Can you verify your original dockerfile on this image?
```
gcr.io/kaniko-project/executor:550
gcr.io/kaniko-project/executor:debug550
```

Thanks a lot in advance!
Tejal

**Comment by tejal29** [comment]
@kferrone This is now fixed by #962. I also verified your dockerfile manually!

Please re-open if this issue persists
